### DISCUSSION TITLE
useNuxtApp().hook not working as expected

### DISCUSSION THREAD

**Discussion by ghost** [nuxt/nuxt#16682] [Questions]
### Environment

- Operating System: `Windows_NT`
- Node Version:     `v16.17.1`
- Nuxt Version:     `3.0.0-rc.11`
- Nitro Version:    `0.5.4`
- Package Manager:  `npm@8.15.0`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

https://stackblitz.com/edit/github-rxxma7?file=composables/useTest.ts

### Describe the bug

I am trying to hook into the Nuxt lifecycle hooks within a composable. However, it doesn't appear to be doing anything. Please can someone provide me with assistance on if I am doing it correctly.

I want to run a fetch function on app:mount / page:start

**Comment by ghost** [comment]
Looked into this further, it seems the only hook that works in a composable is "page:finish", Can someone please explain what the best way to replicate onMounted using Nuxt lifecycle hooks.

**Comment by danielroe** [comment]
It's not reproducible in your reproduction. I suspect the issue is that you are registering the hook after the mounted has already been run.

You can use the following code to debug:

```js
  nuxtApp.hooks.beforeEach(event=> {
    console.log(event.name)
  })
```

I would recommend using `onMounted`.
